src/internal/connector/graph/service.go

```diff
@@ -45,10 +46,15 @@ func (s Service) Client() *msgraphsdk.GraphServiceClient {
 	return s.client
 }
 
-// Seraialize writes an M365 parsable object into a byte array using the built-in
-// application/json writer within the adapter.
 func (s Service) Serialize(object absser.Parsable) ([]byte, error) {
 	writer, err := s.adapter.GetSerializationWriterFactory().GetSerializationWriter("application/json")
+	defer func() {
+		writerErr := writer.Close()
+		if writerErr != nil {
+			err = support.WrapAndAppend("failure during writer closer", writerErr, err)
```

Comment:
This has a gotcha.  even though it updates the `err` value, it won't change the response value.  If you want to return this as an error, you have to name your return values.  Ex: `(bs []byte, err error)`.  Although, imo, that's a bit of an anti-pattern.  I think an info log would be sufficient here.